Lib/ruby/rubyhead.swg

```diff
@@ -110,26 +110,18 @@
  * can be passed as an argument to API functions like Data_Wrap_Struct()
  * and Data_Make_Struct().
  */
```

Comment:
The comment above needs updating. I think the 1st two lines should be replaced with something like:

The following macros are for backwards compatibility and using strictly correct function signatures in C++
when Ruby 2.7 removed ANYARGS and replaced callback function signatures with explicit types.

Is this your understanding as well? Feel free to reword.

Reply:
True, I missed that, but I will have to postpone the change until tomorrow...

Reply:
Fixed it, and learned something about "function declarators with empty parentheses" (https://stackoverflow.com/a/20835620) used in the Ruby C API for C (not C++) extensively before version 2.7. That's why there is this additional condition regarding C++ in the macro definition block.